### PR TITLE
feat: rgb_cmyk_conversion

### PR DESCRIPTION
**Pull request by ali77gh**
# RGB to CMYK conversion

## Description

[RGB](https://en.wikipedia.org/wiki/RGB_color_model) to [CMYK](https://en.wikipedia.org/wiki/CMYK_color_model) Color format algorithm is implemented.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I ran bellow commands using the latest version of **rust nightly**.
- [x] I ran `cargo clippy --all -- -D warnings` just before my last commit and fixed any issue that was found.
- [x] I ran `cargo fmt` just before my last commit.
- [x] I ran `cargo test` just before my last commit and all tests passed.
- [x] I added my algorithm to the corresponding `mod.rs` file within its own folder, and in any parent folder(s).
- [x] I added my algorithm to `DIRECTORY.md` with the correct link.
- [x] I checked `COUNTRIBUTING.md` and my code follows its guidelines.

### CODE REVIEW

File: src/conversions/rgb_cmyk_conversion.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+/// This function Converts RGB to CMYK format
+///
+/// ### Params
+/// * `r` - red
+/// * `g` - green
+/// * `b` - blue
+///
+/// ### Returns
+/// (C, M, Y, K)
+pub fn rgb_to_cmyk(r: u8, g: u8, b: u8) -> (u8, u8, u8, u8) {
```
Comment (vil02) [inline src/conversions/rgb_cmyk_conversion.rs:15]: What do you think about
```suggestion
pub fn rgb_to_cmyk(rgb: (u8, u8, u8)) -> (u8, u8, u8, u8) {
```

File: src/conversions/rgb_cmyk_conversion.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        // red
+        assert_eq!(rgb_to_cmyk(255, 0, 0), (0, 100, 100, 0));
+
+        // green
+        assert_eq!(rgb_to_cmyk(0, 255, 0), (100, 0, 100, 0));
+
+        // blue
+        assert_eq!(rgb_to_cmyk(0, 0, 255), (100, 100, 0, 0));
+    }
```
Comment (vil02) [inline src/conversions/rgb_cmyk_conversion.rs:55]:
```suggestion
    macro_rules! test_rgb_to_cmyk {
        ($($name:ident: $tc:expr,)*) => {
            $(
                #[test]
                fn $name() {
                    let (r, g, b, cmyk) = $tc;
                    assert_eq!(rgb_to_cmyk(r, g, b), cmyk);
                }
            )*
        }
    }

    test_rgb_to_cmyk! {
        white: (255, 255, 255, (0, 0, 0, 0)),
        gray: (128, 128, 128, (0, 0, 0, 49)),
        black: (0, 0, 0, (0, 0, 0, 100)),
        red: (255, 0, 0, (0, 100, 100, 0)),
        green: (0, 255, 0, (100, 0, 100, 0)),
        blue: (0, 0, 255, (100, 100, 0, 0)),
    }
```

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/conversions/rgb_cmyk_conversion.rs (hunk 1 above)
Comment (ali77gh) [reply src/conversions/rgb_cmyk_conversion.rs]: Yeah. It's much better. 
It's easier to pass around a tuple instead of repeating r g b everywhere!

Cool👍

**Comment by ali77gh** [comment]
Ops!
CI failed.
I will make another commit to fix it ASAP.

**Review by vil02 — APPROVED** [review]

**Pull request merged**